src/messaging/ExchangeMgr.cpp

```diff
@@ -376,5 +376,19 @@ void ExchangeManager::CloseAllContextsForDelegate(const ExchangeDelegate * deleg
     });
 }
 
+void ExchangeManager::AbortExchangesForFabricExceptOne(FabricIndex fabricIndex, ExchangeContext * deferred)
+{
+    mContextPool.ForEachActiveObject([&](auto * ec) {
+        if (ec->HasSessionHandle() && ec->GetSessionHandle()->GetPeer().GetFabricIndex() == fabricIndex)
```

Comment:
Why not:
```suggestion
        if (ec->HasSessionHandle() && ec->GetSessionHandle()->GetFabricIndex() == fabricIndex)
```
?  If there's a reason it has to be the more complicated thing, please document that reason so someone does not "fix" it.

I think the main difference is in what happens with a post-AddNOC PASE session.  The code here will _not_ abort exchanges no such a session.  But I think it should abort them...